packages/flutter_tools/test/integration.shard/gen_l10n_test.dart

```diff
@@ -125,8 +131,29 @@ void main() {
       '#l10n 46 (Hello two worlds with 102 total citizens)\n'
       '#l10n 47 ([Hello] -World- #123#)\n'
       '#l10n 48 (Flutter\'s amazing!)\n'
-      '#l10n 49 (Flutter is "amazing"!)\n'
+      '#l10n 49 (Flutter\'s amazing, times 1!)\n'
+      '#l10n 50 (Flutter is "amazing"!)\n'
       '#l10n END\n'
     );
+  }
+
+  test('generated l10n classes produce expected localized strings', () async {
+    setUpAndRunGenL10n();
+    final StringBuffer stdout = await listenToStdOut(() {
+      _flutter.run();
+    });
+    expectOutput(stdout);
+  });
+
+  test('generated l10n classes produce expected localized strings with deferred loading', () async {
+    setUpAndRunGenL10n(args: <String>['--use-deferred-loading']);
+    // When loading the localizations with deferred loading, we will not have
+    // the localizations the first build, thus we run the app and reload it once
+    // to get the output.
+    await _flutter.run();
+    final StringBuffer stdout = await listenToStdOut(() {
```

Comment:
I think you should have an expect to check stdout before you do the deferred loading as well.